src/sources/util/http.rs

```diff
@@ -0,0 +1,113 @@
+use crate::event::Event;
+use bytes::Buf;
+use futures::{sync::mpsc, Future, Sink};
+use serde::Serialize;
+use std::error::Error;
+use std::fmt::{self, Display};
+use std::net::SocketAddr;
+use std::sync::{Arc, Mutex};
+use stream_cancel::Tripwire;
+use warp::filters::BoxedFilter;
+use warp::http::{HeaderMap, StatusCode};
+use warp::{Filter, Rejection};
+
+#[derive(Serialize, Debug)]
+pub struct ErrorMessage {
+    code: u16,
+    message: String,
+}
+impl ErrorMessage {
+    pub fn new(code: StatusCode, message: String) -> Self {
+        ErrorMessage {
+            code: code.as_u16(),
+            message,
+        }
+    }
+}
+impl Error for ErrorMessage {}
+impl Display for ErrorMessage {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}: {}", self.code, self.message)
+    }
+}
+
+fn string_to_static_str(s: String) -> &'static str {
+    //necessary because warp 0.1.18 needs a &'static str for the path
+    Box::leak(s.into_boxed_str())
```

Comment:
I don't think this is safe, and it definitely looks like a memory leak. https://doc.rust-lang.org/std/boxed/struct.Box.html#method.leak

Reply:
Yeah it's kinda gross. Couldn't see a way around this if we wanted the path to come from the config, since the version of warp needs &'static str.

If there's a way around this that I don't know then I'd love to know it.

Or if it's more important to not have the leak, we could just pass in a &'static str and have it be hard-coded.

Reply:
I'd prefer to see it hard coded until we upgrade warp to 0.2.x, since this problem only exists for us in 0.1.x. :)

If we're listening on a socket, I suggest just listen to everything on that socket. The path could be part of the event maybe? @binarylogic 's spec in #328 didn't mention anything about being able to set paths.

Reply:
OK! - I think I'll just make it hard-coded for now, since it doesn't seem to be needed.

You're right, there is to be a way to get the full-path after the fact ([see here](https://docs.rs/warp/0.1.18/warp/filters/path/fn.full.html)) but they say 

> should probably not be used for request matching/routing.

so not sure about that, better I think to have the hard-coded path and worry about this when someone wants a configurable path (by which point we'll hopefully be on warp 0.2.x!)